### PR TITLE
Add userComment field in SampleTimelineAuditEvent and in the SampleTimelineEvent metadata

### PR DESCRIPTION
**Pull request by labkey-susanh**
#### Rationale
We are adding the ability for users to provide comments when editing and deleting samples and need to capture these comments in the audit logs.  We already have the mechanism for getting these comments, just need to pass them through to the sample-specific operations.

#### Related Pull Requests
- https://github.com/LabKey/sampleManagement/pull/1518
- https://github.com/LabKey/inventory/pull/676

#### Changes
* Add UserComment field to SampleTimelineAuditEvent.
* pass user comment through to various methods that create audit logs

### CODE REVIEW

File: experiment/src/org/labkey/experiment/api/SampleTypeServiceImpl.java [hunk 1]
```diff
@@ -1082,14 +1082,14 @@ public String getCommentDetailed(QueryService.AuditAction action, boolean isUpda
     public DetailedAuditTypeEvent createDetailedAuditRecord(User user, Container c, AuditConfigurable tInfo, QueryService.AuditAction action, @Nullable String userComment, @Nullable Map<String, Object> row, Map<String, Object> existingRow)
     {
         // not doing anything with userComment at the moment
```
Comment (cnathe) [inline experiment/src/org/labkey/experiment/api/SampleTypeServiceImpl.java]: this comment can now be removed right?

File: experiment/src/org/labkey/experiment/api/SampleTypeServiceImpl.java [hunk 2]
```diff
@@ -1082,14 +1082,14 @@ public String getCommentDetailed(QueryService.AuditAction action, boolean isUpda
...
     {
         // not doing anything with userComment at the moment
-        return createAuditRecord(c, getCommentDetailed(action, !existingRow.isEmpty()), action, row, existingRow);
+        return createAuditRecord(c, getCommentDetailed(action, !existingRow.isEmpty()), userComment, action, row, existingRow);
     }
 
     @Override
     protected AuditTypeEvent createSummaryAuditRecord(User user, Container c, AuditConfigurable tInfo, QueryService.AuditAction action, @Nullable String userComment, int rowCount, @Nullable Map<String, Object> row)
     {
         // not doing anything with userComment at the moment
```
Comment (cnathe) [inline experiment/src/org/labkey/experiment/api/SampleTypeServiceImpl.java]: this comment can now be removed right?

**Review by cnathe — APPROVED** [review]

File: experiment/src/org/labkey/experiment/api/SampleTypeServiceImpl.java (hunk 2 above)
Comment (labkey-susanh) [reply experiment/src/org/labkey/experiment/api/SampleTypeServiceImpl.java]: Done

File: experiment/src/org/labkey/experiment/api/SampleTypeServiceImpl.java (hunk 1 above)
Comment (labkey-susanh) [reply experiment/src/org/labkey/experiment/api/SampleTypeServiceImpl.java]: Done.

**Pull request merged**